tradefed/__init__.py

```diff
@@ -5,18 +5,103 @@
 import xmlrpc
 import yaml
 import xml.etree.ElementTree as ET
+from celery import chord as celery_chord
 from io import BytesIO
 from django.conf import settings
+from django.db import transaction
 from squad.plugins import Plugin as BasePlugin
 from urllib.parse import urljoin
+from squad.celery import app as celery
 from squad.core.utils import join_name
-from squad.core.models import Suite, SuiteMetadata, Test, KnownIssue, Status
+from squad.core.models import Suite, SuiteMetadata, Test, KnownIssue, Status, TestRun, ProjectStatus
 from squad.core.tasks import get_suite
 
 
 logger = logging.getLogger()
 
 
+@celery.task
+def update_build_status(results_list, testrun_id):
```

Comment:
I don't think you need this function, hence updating project status will be done right after `prostprocess_testjob`: https://github.com/Linaro/squad/blob/master/squad/ci/models.py#L138

Reply:
This is correct when all postprocessing happens in the same task. It's not the case any more as I'm creating more tasks and the main task exits. In case of the "big" CTS job refresh or build status happens at around 10% of CTS tests processed. Status update happens twice but I think this is the best we can do.

Reply:
Oh, I thought `celery_chord` was a blocking function, this is necessary then!

I also thought about a scenario, though rare might happen every once in a while: 

`UpdateProjectStatus` will call `maybe_notify_project_status` and a email notification might get sent with incomplete results.

Should we worry about that? I think this is OK

Reply:
This is a possibility. I'm not sure how to prevent it right now.